ts/sagas/startup/loadVisibleServicesHandler.ts

```diff
@@ -118,9 +120,30 @@ export function* loadVisibleServicesRequestHandler(
           );
         })
         .map(_ => _.service_id);
-
-      // Parallel fetch of those services that we haven't loaded yet or need to be updated
+      // Parallel fetch of those services content that we haven't loaded yet or need to be updated
       yield all(serviceIdsToLoad.map(id => put(loadService.request(id))));
+
+      // Check the metadata of what services should be loaded
+      // Metadata has no version and they are updated only if:
+      // - service content version is updated ( after loadService.request service metadata are loaded again)
+      // - service metadata stored on redux store has error state (meaning somthing wrong when loaded)
+      const servicesMetadataById = yield select(servicesMetadataByIdSelector);
+      const serviceIdsToLoad2 = visibleServices
+        .filter(service => {
+          const serviceId = service.service_id;
+          const storedMetadata = servicesMetadataById[serviceId];
+          return (
+            // Service load never occurs or it ends up with an error
+            serviceIdsToLoad.indexOf(serviceId) === -1 &&
+            !pot.isLoading(storedMetadata) &&
+            (pot.isError(storedMetadata) || pot.isNone(storedMetadata))
+          );
+        })
+        .map(_ => _.service_id);
+      // Parallel fetch of those services metadata that we haven't loaded yet or need to be updated
+      yield all(
```

Comment:
You could remove previous loading request and merge into only one (I suggest also a filtering becasue you could have same id in both arrays)
```typescript
// this ensure no id duplication (please give me a better name :D )
const serviceIdsToLoad3 = serviceIdsToLoad
        .concat(serviceIdsToLoad2)
        .filter((v, i, a) => a.indexOf(v) === i);
// Parallel fetch of those services metadata that we haven't loaded yet or need to be updated
      yield all(
        serviceIdsToLoad3.map(id =>
          put(contentServiceLoad.request(id))
        )
      );
```

Reply:
Are you meaning to use `serviceIdsToLoad3` instead of both `serviceIdsToLoad1` and `serviceIdsToLoad2`?  
If yes, they are expected to do different uploads: 
`serviceIdsToLoad1` --> service content load refresh
`serviceIdsToLoad2` --> service metadata load/refresh
It could happen that, for a service, its content is properly loaded while its metadata is not.

Reply:
@valentinacala 
Yes! you're right! I thought they raise the same action, indeed one is **loadService** and the other one is **contentServiceLoad**
btw Could we change from ** serviceIdsToLoad2** to **serviceContentsToLoad**?